internal/provider/waypoint/resource_waypoint_action.go

```diff
@@ -0,0 +1,629 @@
+							"method": schema.StringAttribute{
+								Description: "The HTTP method to use for the request.",
+								Required:    true,
+							},
+							"headers": schema.MapAttribute{
+								Description: "Key value headers to send with the request.",
+								Optional:    true,
+								ElementType: types.StringType,
+							},
+							"url": schema.StringAttribute{
+								Description: "The full URL this request should make when invoked.",
+								Optional:    true,
+							},
+							"body": schema.StringAttribute{
+								Description: "The body to be submitted with the request.",
+								Optional:    true,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *ActionResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *ActionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan *ActionResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+
+	client := r.client
+	ns, err := getNamespaceByLocation(ctx, client, loc)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"error getting namespace by location",
+			err.Error(),
+		)
+		return
+	}
+
+	modelBody := &waypoint_models.HashicorpCloudWaypointWaypointServiceCreateActionConfigBody{
+		ActionConfig: &waypoint_models.HashicorpCloudWaypointActionConfig{
+			Request: &waypoint_models.HashicorpCloudWaypointActionConfigRequest{},
+		},
+	}
+
+	modelBody.ActionConfig.Name = plan.Name.ValueString()
+
+	if !plan.Description.IsUnknown() {
+		modelBody.ActionConfig.Description = plan.Description.ValueString()
+	}
+
+	if !plan.ActionURL.IsUnknown() {
+		modelBody.ActionConfig.ActionURL = plan.ActionURL.ValueString()
+	}
+
+	var diags diag.Diagnostics
+
+	// This is a proxy for the request type, as Custom.Method is required for Custom requests
+	if !plan.Request.Custom.Method.IsUnknown() && !plan.Request.Custom.Method.IsNull() {
+		modelBody.ActionConfig.Request.Custom = &waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustom{}
+
+		method, err := convertMethodToEnumType(plan.Request.Custom.Method.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError(
```

Comment:
Should we include the `err` too here?

Reply:
The error that is returned is pretty redundant here, it is just a less verbose version of the error that is already added so I would say not in this case.